### DISCUSSION TITLE
Tasmota Shelly1 Light over PIR on SW in and over Tasmota APP

### DISCUSSION THREAD

**Discussion by mischa-t** [arendst/Tasmota#20158] [Q&A]
Hello, im trying to build an Light for Outside. I allready managed to build it with a Shelly1 and and PIR which works good, i can control the Light over the PIR or over the Tasmota APP.
The problem that i have is that i want that when i switch the Light on over the app that the PIR has no input.
At this point when i seat outside and i switch on the light over the app or the pir the light goes out, can i make the app as a master switch over the shelly sw inut?

**Comment by barbudor** [comment]
May be you can elaborate a little more on how you make the PIR control the Shelly, what is your Tasmota configuration (Rules ? Switch mode ?) and what does that app is doing (Tasmota does NOT have an APP, only on web interface. Any "app" is not coming from here and we don't support it)

**Comment by mischa-t** [comment]
Sorry, i mean the Web Interface.
As a Parameter i use Shelly 1(46) and i have GPIO4 as relay1 and GPIO5 as Switch_n1.
When the Output of the PIR goes to GPIO5 it turns the light on.
{"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}

**Reply by barbudor** [reply]
So everytime you move in front of the PIR, the light goes ON, OFF, ON, OFF ?

**Comment by mischa-t** [comment]
I configured the Input of the Shelly1 as a button and not a switch. The PIR has a Timer, so as long as the PIR is active the Light is on.
The Priority should be the On/Off Switch in the Webinterface so that when i turn the light on over the interface it will not go on and off by movement.

**Reply by mischa-t** [reply]
Oh i have the following switchmode active:
SwitchMode1":0

**Comment by mischa-t** [comment]
So i managed to get it running with the following:

switchmode1 1
Setoption114 1

and 3 Rules:

Rule1
on Switch2#state=1 do backlog event checksunrise=%time%; event checksunset=%time% endon  
on event#checksunrise<%sunrise% do power1 1 endon   
on event#checksunset>%sunset% do power1 1 endon
Rule2
on switch2#state do backlog power2 %value%; power1 1; ruletimer1 60 endon on rules#timer=1 do power1 0 endon
Rule3
on power3#state=1 do backlog power1 1; rule1 0, rule2 0 endon       
on power3#state=0 do backlog power1 0; rule1 1, rule2 1 endon

Rules 1,2 & 3 are set to on.

At this point when the PIR (output 220V on SW in from Shelly1) the Light goes on for 60 seconds (acc to Rule 2).
When Webswitch3 is pressed the light goes on and Rules 1 & 2 are set inoperativ.

These 2 things work, i only have a problem with Rule1 that the PIR is only active before sunrise and after sunset.

Does anybody know what i am doing wrong?

**Reply by sfromis** [reply]
Read the console (after having `Weblog 3`) for details when it does not behave like you want.

And I noticed that you're not mentioning what SwitchMode you have for Switch2, or even if you have done something there.

**Reply by mischa-t** [reply]
Hello sfromis,
Your right, i forgot setting switchmode2 1, it was default on 0.
What do i see with the consol command Weblog 3, tryed that but only this happend:

18:07:17.224 CMD: weblog 3
18:07:17.226 SRC: WebConsole from 192.168.178.24
18:07:17.229 CMD: Grp 0, Cmd 'WEBLOG', Idx 1, Len 1, Pld 3, Data '3'
18:07:17.233 RSL: RESULT = {"WebLog":3}

**Reply by sfromis** [reply]
And **after** having set the logging level, you start to get more details when Tasmota does something.

**Reply by mischa-t** [reply]
Hello sfromis,
i have still the effect that before sunset the light goes on.
Here are the logs:

16:44:20.588 RUL: SWITCH2#STATE=1 performs "backlog event checksunrise=1004; event checksunset=1004"
16:44:20.591 SRC: Rule
16:44:20.594 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 47, Pld -99, Data 'event checksunrise=1004; event checksunset=1004'
16:44:20.620 RUL: SWITCH2#STATE performs "backlog power2 1; power1 1; ruletimer1 60"
16:44:20.622 SRC: Rule
16:44:20.624 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 33, Pld -99, Data 'power2 1; power1 1; ruletimer1 60'
16:44:20.637 RSL: RESULT = {"Switch2":{"Action":"ON"}}
16:44:20.659 SRC: Backlog
16:44:20.661 CMD: Grp 0, Cmd 'EVENT', Idx 1, Len 17, Pld -99, Data 'checksunrise=1004'
16:44:20.665 RSL: RESULT = {"Event":"Done"}
16:44:20.879 SRC: Backlog
16:44:20.882 CMD: Grp 0, Cmd 'EVENT', Idx 1, Len 16, Pld -99, Data 'checksunset=1004'
16:44:20.887 RSL: RESULT = {"Event":"Done"}
16:44:20.931 RUL: EVENT#CHECKSUNSET>%SUNSET% performs "power1 1"
16:44:20.933 SRC: Rule
16:44:20.936 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 1, Pld 1, Data '1'
16:44:20.940 RSL: RESULT = {"POWER1":"ON"}
16:44:20.943 RSL: POWER1 = ON

Hope you can see something here, for me 50% is chinese

**Reply by sfromis** [reply]
Well, it looks like it is working perfectly, as it is **after** sunset according to the time zone settings and geographical lat/lon configured. To have correct settings, I can recommend this tool:
https://tasmotatimezone.com/

Using `Status 7`, you can check what sunrise/sunset times Tasmota calculated. At 16.44 the sun has set in northern Europe, including the default Paris lat/lon (if you did not configure properly).

Using the command `Sunrise`, you can also select your twilight adjustment, allowing you to extend the time considered daylight to before/after the sun crosses the horizon.

**Reply by mischa-t** [reply]
The Sunset and Sunrise time fit, the Lang and Lat are also ok.
Is then my Rule wrong or my settings that the Light turns on if the PIR see's movement after sunrise and before sunset.
Everything else works ok only that the light turns on daysover.

**Reply by sfromis** [reply]
Using a test rule with the same trigger, and Event commands from the console, I found that a number up to and including the `%Sunset%` minute did not trigger, while a number one minute past this caused the trigger to fire. You can check yourself
`on Event#CheckSunset>%Sunset% do var1 Sunset hit. %value% > %sunset% endon`

And then commands like `Event CheckSunset=1004` to match what you posted. Or other numbers to see what causes it to fire.

**Reply by sfromis** [reply]
Note that the daylight time calculations are not trying to adjust for altitude variations (small factor), or landscape features (like high terrain near lower), which can move when there can be line-of-sight to the sun. And north of the polar circle, it will not work all year around.

**Reply by mischa-t** [reply]
So it seams to work now, i adjusted rule1 again. The light stays of when the PIR acitvated. I just have to check it after sunset if it works.
I just dont understand the numbers and data in the CMD line like sunset and sunrise "Number".
I copied the consol data:

10:17:21.762 RUL: SWITCH2#STATE=1 performs "backlog event checksunrise=617; event checksunset=617"
10:17:21.765 SRC: Rule
10:17:21.767 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 45, Pld -99, Data 'event checksunrise=617; event checksunset=617'
10:17:21.789 RSL: RESULT = {"Switch2":{"Action":"ON"}}
10:17:21.811 SRC: Backlog
10:17:21.814 CMD: Grp 0, Cmd 'EVENT', Idx 1, Len 16, Pld -99, Data 'checksunrise=617'
10:17:21.818 RSL: RESULT = {"Event":"Done"}
10:17:22.063 SRC: Backlog
10:17:22.066 CMD: Grp 0, Cmd 'EVENT', Idx 1, Len 15, Pld -99, Data 'checksunset=617'
10:17:22.070 RSL: RESULT = {"Event":"Done"}
10:17:57.342 RSL: RESULT = {"Switch2":{"Action":"OFF"}}

**Reply by sfromis** [reply]
The numbers are simply minutes past midnight, `hours * 60 + minutes`. Like when 10 hours and 17 minutes becomes 617.

**Reply by mischa-t** [reply]
i understand, quite easy.

**Reply by mischa-t** [reply]
Now everything seems to work, i readjusted the sunset/sunrise and combined the code. Thanks for the Help:

Rule1
on Switch2#state=1 do backlog event checksunrise=%time%; event checksunset=%time% endon   
on event#checksunrise<%sunrise% do power1 1 endon   
on event#checksunset>%sunset% do backlog power1 1; ruletimer1 120 endon   
on rules#timer=1 do Power1 OFF endon

Rule2
on power3#state=1 do backlog rule1 0; power1 1 ENDON
on power3#state=0 do backlog rule1 1; power1 0 ENDON

**Comment by Noschvie** [comment]
@mischa-t thanks for sharing the rules. But how is the configuration of the device to reproduce your complete setup?
Thanks!

**Reply by mischa-t** [reply]
Hello Noschvie, as above. Its a shelly1 with a PIR (220V) output to the SW Input of the Shelly and a Lamp on the Output of the Shelly.
![image](https://github.com/arendst/Tasmota/assets/105169614/29a0479f-1cd8-4171-b3be-d49cdb159277)
![image](https://github.com/arendst/Tasmota/assets/105169614/0fe870c8-adaf-4e94-b687-eb90df88d89e)
switchmode1 1
switchmode2 1
Setoption114 1

**Comment by mischa-t** [comment]
It seems to be that I was to early with my happiness, I thought it would work but there ist still a little flaw inside. 
The thing with sunset and sunrise works but after sunset when movement is sensed the light goes on, this is ok. After 120s the light should go off but it doesn't. I can't find the fault in rule1.

**Reply by sfromis** [reply]
Again, watching the log for details is the key to following what happens in Tasmota.

**Reply by mischa-t** [reply]
Got the Log files. 
Found that the Ruletimer is working before midnight, after midnight the light stays on.

23:03:26.797 WIF: Checking connection...
23:03:39.916 NTP: Sync time...
23:03:39.919 WIF: DNS resolved 'ptbtime3.ptb.de' (192.53.103.103) in 1 ms
23:03:40.086 RTC: UTC 2024-01-05T22:03:40, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
23:03:40.000 RTC: Synced by NTP

23:25:52.682 RUL: SWITCH2#STATE=1 performs "backlog event checksunrise=1405; event checksunset=1405"
23:25:52.685 SRC: Rule
23:25:52.688 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 47, Pld -99, Data 'event checksunrise=1405; event checksunset=1405'
23:25:52.714 RSL: RESULT = {"Switch2":{"Action":"ON"}}
23:25:52.742 SRC: Backlog
23:25:52.745 CMD: Grp 0, Cmd 'EVENT', Idx 1, Len 17, Pld -99, Data 'checksunrise=1405'
23:25:52.750 RSL: RESULT = {"Event":"Done"}
23:25:52.957 SRC: Backlog
23:25:52.959 CMD: Grp 0, Cmd 'EVENT', Idx 1, Len 16, Pld -99, Data 'checksunset=1405'
23:25:52.964 RSL: RESULT = {"Event":"Done"}
23:25:53.014 RUL: EVENT#CHECKSUNSET>%SUNSET% performs "backlog power1 1; ruletimer1 120"
23:25:53.016 SRC: Rule
23:25:53.018 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 24, Pld -99, Data 'power1 1; ruletimer1 120'
23:25:53.038 SRC: Backlog
23:25:53.041 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 1, Pld 1, Data '1'
23:25:53.044 RSL: RESULT = {"POWER1":"ON"}
23:25:53.046 RSL: POWER1 = ON
23:25:53.289 SRC: Backlog
23:25:53.292 CMD: Grp 0, Cmd 'RULETIMER', Idx 1, Len 3, Pld 120, Data '120'
23:25:53.298 RSL: RESULT = {"T1":120,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
23:25:54.214 CFG: Saved to flash at FA, Count 10405, Bytes 4096
23:25:58.436 WIF: Checking connection...
23:26:04.633 RSL: RESULT = {"Switch2":{"Action":"OFF"}}
23:27:53.963 RUL: RULES#TIMER=1 performs "Power1 OFF"
23:27:53.965 SRC: Rule
23:27:53.967 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 3, Pld 0, Data 'OFF'
23:27:53.972 RSL: RESULT = {"POWER1":"OFF"}
23:27:53.974 RSL: POWER1 = OFF
23:27:54.816 CFG: Saved to flash at F9, Count 10406, Bytes 4096

00:03:25.939 WIF: Sending Gratuitous ARP
00:03:39.924 NTP: Sync time...
00:03:40.136 WIF: DNS resolved 'ptbtime3.ptb.de' (192.53.103.103) in 209 ms
00:03:40.254 RTC: UTC 2024-01-05T23:03:41, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
00:03:41.000 RTC: Synced by NTP

00:17:39.361 RUL: SWITCH2#STATE=1 performs "backlog event checksunrise=17; event checksunset=17"
00:17:39.363 SRC: Rule
00:17:39.366 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 43, Pld -99, Data 'event checksunrise=17; event checksunset=17'
00:17:39.393 RSL: RESULT = {"Switch2":{"Action":"ON"}}
00:17:39.421 SRC: Backlog
00:17:39.423 CMD: Grp 0, Cmd 'EVENT', Idx 1, Len 15, Pld -99, Data 'checksunrise=17'
00:17:39.427 RSL: RESULT = {"Event":"Done"}
00:17:39.471 RUL: EVENT#CHECKSUNRISE<%SUNRISE% performs "power1 1"
00:17:39.473 SRC: Rule
00:17:39.476 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 1, Pld 1, Data '1'
00:17:39.480 RSL: RESULT = {"POWER1":"ON"}
00:17:39.483 RSL: POWER1 = ON
00:17:39.708 SRC: Backlog
00:17:39.711 CMD: Grp 0, Cmd 'EVENT', Idx 1, Len 14, Pld -99, Data 'checksunset=17'
00:17:39.715 RSL: RESULT = {"Event":"Done"}
00:17:40.091 CFG: Saved to flash at F8, Count 10407, Bytes 4096
00:17:54.776 WIF: Checking connection...
00:17:55.466 RSL: RESULT = {"Switch2":{"Action":"OFF"}}

**Reply by mischa-t** [reply]
I think i found the Problem why the Light does not go off after Midnight.
In the Rule 1 i added the following to  event#checksunrise.

Rule1
  on Switch2#state=1 do backlog event checksunrise=%time%; event checksunset=%time% endon    
  on event#checksunrise<%sunrise% do backlog power1 1; **ruletimer1 120** endon    
  on event#checksunset>%sunset% do backlog power1 1; ruletimer1 120 endon    

I'll check that after midnight.

**Reply by sfromis** [reply]
You can test that before midnight by issuing the Event commands manually, with any minute value to see what happens.

**Reply by mischa-t** [reply]
how do i issue the commands manually?

**Reply by sfromis** [reply]
Type them in the console, like `Event checksunrise=1405`
The same as in the rules, just that you can't use anything with `%` on the command line.

**Reply by mischa-t** [reply]
So, i managed to soll it,
on event#checksunrise<%sunrise% do backlog power1 1; ruletimer1 120 endon
on event#checksunset>%sunset% do backlog power1 1; ruletimer1 120 endon
in both events i need the ruletimer1, thats why it worked until 23:59 and after 00:01 it didnt.

Thank for the Help
